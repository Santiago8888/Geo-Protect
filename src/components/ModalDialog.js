import React from "react";
import PropTypes from 'prop-types';

const ModalDialog =  props => {
  const header = props.title ? (
    <div className='modal-card-head has-background-white'>
      {props.title}
    </div>
  ) : null;
  const footer = props.footer ? (
    <footer className='modal-card-foot'>{ props.footer }</footer>
  ) : null;
  return (
    <div className={`modal ${props.className} ${props.isActive ? 'is-active': ''}`}>
      <div className="modal-background"/>
      <div className="modal-card">
        <React.Fragment>
          { header }
          <button className="modal-close is-large"
                  aria-label="close"
                  onClick={
                    e => {
                      if (
                        props.hasOwnProperty('onClose') &&
                        typeof props.onClose !== 'undefined'
                      ) {
                        props.onClose(e);
                      }
                    }
                  }
          >
            &times;
          </button>
        </React.Fragment>
        <div className='modal-card-body'>
          { props.children }
        </div>
        { footer }
      </div>
    </div>
  );
};

ModalDialog.propTypes ={
  title: PropTypes.node,
  className: PropTypes.string,
  isActive: PropTypes.bool,
  footer: PropTypes.node,
  onClose: PropTypes.func,
};

export default ModalDialog;