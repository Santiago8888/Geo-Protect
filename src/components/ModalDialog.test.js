import React from "react";
import ModalDialog from "./ModalDialog"
import {mount} from 'enzyme';

describe('ModalDialg tests', () => {
  it('should render successfully using the passed in child component', () => {
    const modal = (
      <ModalDialog>
        <h1>Hello World</h1>
      </ModalDialog>
    );
    const sut = mount(modal);
    expect(sut.find('.modal')).toBeDefined();
    expect(sut.find('.modal-card-body')).toBeDefined();
    expect(sut).toMatchSnapshot();
  });

  it('should render a header when passed in as prop', () => {
    expect(
      <ModalDialog title='This is a test'>
        <h1>Hello World</h1>
      </ModalDialog>
    ).toMatchSnapshot();
  })
});