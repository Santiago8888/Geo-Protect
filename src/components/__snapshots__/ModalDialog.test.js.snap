// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialg tests should render a header when passed in as prop 1`] = `
<_default
  title="This is a test"
>
  <h1>
    Hello World
  </h1>
</_default>
`;

exports[`ModalDialg tests should render successfully using the passed in child component 1`] = `
<_default>
  <div
    className="modal undefined "
  >
    <div
      className="modal-background"
    />
    <div
      className="modal-card"
    >
      <button
        aria-label="close"
        className="modal-close is-large"
        onClick={[Function]}
      >
        Ã—
      </button>
      <div
        className="modal-card-body"
      >
        <h1>
          Hello World
        </h1>
      </div>
    </div>
  </div>
</_default>
`;
